RELEASING.rst

```diff
@@ -0,0 +1,61 @@
+Creating a release
+==================
+
+This document is documentation intended for Buildbot maintainers. It documents the release
+process of Buildbot.
+
+Step 1: Release notes PR
+------------------------
+
+Open a new branch (e.g. `release`) and run the following:
+
+    . .venv/bin/activate && make release_notes VERSION=x.y.z
+
+This collects the release notes using the `towncrier` tool and then commits the result.
+This step is done as a PR so that CI can check for spelling errors and similar issues. Local checks
+are insufficient as spelling check in particular depends on what dictionaries are installed.
+
+It's best to run `make docs` afterwards and check `master/docs/_build/html/relnotes/index.html`
+file for obvious rendering errors. This will have much faster turnaround compared to if the
+error is noticed after the CI runs. If any errors are found, just amend the commit created by
+`make release_notes`.
+
+Step 2: Merge the release notes PR
+----------------------------------
+
+Step 3: Perform actual release
+------------------------------
+
+This step requires the Buildbot git repository to contain `buildbot` remote that points to
+https://github.com/buildbot/buildbot and can be pushed to.
+
+Additionally, the Buildbot docs repository (https://github.com/buildbot/bbdocs) must be checked
+out at `../bbdocs` path.
+
+Pull the merge commit created on the `master` branch during the step 2. Then run:
+
+    make release VERSION=x.y.z
+
+This will create the required tags, make documentation, copy it to bbdocs repo and push everything.
+
+Step 4: Wait for CircleCi to create release tarballs
+----------------------------------------------------
+
+The push of tags created during step 3 will activate CircleCi configuration that generates tarballs
+and uploads them to Github. This is a good time to draft a release on the Github UI
```

Comment:
Elsewhere you have "GitHub" so we should be consistent with casing everywhere and change this to "GitHub" too.